03-rnaseq/clustering/clustering_heatmap_example.Rmd

```diff
@@ -0,0 +1,220 @@
+file and a data tsv file. This chunk of code will read the both tsv files and 
+add them as data.frame objects to your environment.
+
+The data that we downloaded from refine.bio for this analysis is a dataset, with
+a total of four samples, containing RNA-sequencing results for mouse primary
+natural killer (NK) cells and NK cells treated with DKK2 for 24 hours.
+
+You can download this dataset by going to the following link and clicking 
+`Download Now`: https://www.refine.bio/experiments/SRP110235/mouse-natural-killer-cells-response-to-dkk2-treatment
+
+```{r Import data from .tsv files}
+# Read in metadata tsv file
+metadata <- readr::read_tsv(file.path(
+  "data", # Replace with path to your metadata file
+  "metadata_SRP110235.tsv" # Replace with the name of your metadata file
+)) %>%
+  dplyr::select(refinebio_accession_code, refinebio_title)
+
+# Read in data tsv file
+df <- readr::read_tsv(file.path(
+  "data", # Replace with path to your data file
+  "SRP110235.tsv" # Replace with the name of your data file
+)) %>%
+  tibble::column_to_rownames("Gene")
+```
+
+Let's ensure that the metadata and data are in the same sample order. 
+
+```{r}
+# Make the data in the order of the metadata
+df <- df %>% dplyr::select(metadata$refinebio_accession_code)
+
+# Check if this is in the same order
+all.equal(colnames(df), metadata$refinebio_accession_code)
+```
+
+## 3) Choose genes of interest
+Although you may want to create a heatmap including all of the genes in the set,
+alternatively, the heatmap could be created using only genes of interest. 
+For this example, we will sort genes by variance, but there are many alternative
+criterion by which you may want to sort your genes <i>eg</i> fold change,
+t-statistic, membership to a particular gene ontology, so on. 
+
+```{r Choose genes}
+# Calculate the variance for each gene
+variances <- apply(df, 1, var)
+
+# Determine summary statistics for gene variances
+sum.stats.var <- summary(variances)
+
+# Subset the data choosing only genes whose variances are in the upper quartile
+df.by.var <- df[which(variances > sum.stats.var[5]), ]
+```
+
+## 4) Create a DESeq dataset
+A `DESeqDataSet` object is required to carry out the normalization step that
+you will see later using a function from the `DESeq2` package. This chunk of
+code will turn the data.frame (or matrix) into a `DESeqDataSet` object.
+
+```{r Create a DESeq dataset}
+# The `DESeqDataSetFromMatrix` function needs the values to be converted to
+# integers
+df.by.var <- apply(df.by.var, 2, as.integer)
+
+# Create a `DESeqDataSet` object
+dds <- DESeqDataSetFromMatrix(
+  countData = df.by.var,
+  colData = metadata,
+  design = ~refinebio_title
+)
+
+# Place the colData in a separate data.frame object for annotation
+col_df <- as.data.frame(colData(dds))
+
+# Define a `Groups` column variable
+col_df <- col_df %>%
+  dplyr::mutate(Groups = rep(c("SP", "MP"), # Replace with group names relevant to your data
```

Comment:
In other words, can you explain why you used this `colData(dds)` strategy you are using here? It isn't clear to me, but this could because I haven't tried this myself.

Reply:
I will give your first suggestion a try and let you know from there. I did try something similar before and that did not work, but I believe it was due to the way I generated the `Groups` column in that case.